core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactorySpec.js

```diff
@@ -0,0 +1,57 @@
+describe('NumberWithUnitsObjectFactory', function() {
+  beforeEach(module('oppia'));
+
+  describe('number with units object factory', function() {
+    var NumberWithUnits = null;
+    var Units = null;
+    var Fraction = null;
+
+    beforeEach(inject(function($injector) {
+      NumberWithUnits = $injector.get('NumberWithUnitsObjectFactory');
+      Units = $injector.get('UnitsObjectFactory');
+      Fraction = $injector.get('FractionObjectFactory');
+    }));
+
+    it('should convert units to dict format', function() {
+      expect(new Units('kg / kg^2 K mol / (N m s^2) K s').toDict()).toEqual(
+        {kg: -1, K: 2, mol: 1, N: -1, m: -1, s: -1});
+      expect(new Units('mol/(kg / (N m / s^2)').toDict()).toEqual(
+        {mol: 1, kg: -1, N: 1, m: 1, s: -2});
+    });
+
+    it('should convert units from dict to string format', function() {
+      expect(Units.fromDictToString(
+        {kg: -1, K: 2, mol: 1, N: -1, m: -1, s: -1})).toBe(
+          'kg^-1 K^2 mol^1 N^-1 m^-1 s^-1');
+      expect(Units.fromDictToString(
+        {mol: 1, kg: -1, N: 1, m: 1, s: -2, K: -1})).toBe(
+          'mol^1 kg^-1 N^1 m^1 s^-2 K^-1');
+    });
+
+    it('should parse valid units strings', function() {
```

Comment:
What is the purpose of this test? Seems like it is comparing output against the same function.

Reply:
The purpose of this test is to check the dict of units formed with Units object and dict formed after parsing units from 'fromRawInputString'.